pkg/apis/serving/v1beta1/component.go

```diff
@@ -120,6 +120,11 @@ type ComponentExtensionSpec struct {
 	// The deployment strategy to use to replace existing pods with new ones. Only applicable for raw deployment mode.
 	// +optional
 	DeploymentStrategy *appsv1.DeploymentStrategy `json:"deploymentStrategy,omitempty"`
+	// Specifies an application protocol.
+	// https://kubernetes.io/docs/concepts/services-networking/service/#application-protocol
+	// +kubebuilder:validation:Enum=kubernetes.io/h2c
+	// +optional
+	AppProtocol *string `json:"appProtocol,omitempty"`
```

Comment:
is this expected to have more values?
I think it is just to use http2 with services, right?
Isn't better to add a flag instead?